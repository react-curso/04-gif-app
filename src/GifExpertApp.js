import React, { useState } from 'react'
import {AddCategory} from './components/AddCategory';
import {GifGrid} from './components/GifGrid';

export const GifExpertApp = ( {defaultCategories = []} ) => {
     
      const [categories, setCategories] = useState( defaultCategories );
     
     //      setCategories( c => [...c, 'Goku'] );

     return (
          <>
               <h2 id="title">GifExpertApp</h2>

               <AddCategory categories={ setCategories } />
               
               <hr />
                    
               <ol>
                    {
                         categories.map( c => {
                              return <GifGrid key={c} category={c} />
                         })
                    }
               </ol>

          </>
     )
}


