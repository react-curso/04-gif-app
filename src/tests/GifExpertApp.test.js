import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';

describe('test component GifExpertApp', () => {
    
    test('renderer component', () => {
        
        const wrapper = shallow( <GifExpertApp /> )
        expect( wrapper ).toMatchSnapshot();

    })
    
    test('debe de mostrar lista de categorias ', () => {
        
        const categories = ['goku', 'homero'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );


    })
    

})
