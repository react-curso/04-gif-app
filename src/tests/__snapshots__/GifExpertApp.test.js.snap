// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component GifExpertApp debe de mostrar lista de categorias  1`] = `
<Fragment>
  <h2
    id="title"
  >
    GifExpertApp
  </h2>
  <AddCategory
    categories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="goku"
      key="goku"
    />
    <GifGrid
      category="homero"
      key="homero"
    />
  </ol>
</Fragment>
`;

exports[`test component GifExpertApp renderer component 1`] = `
<Fragment>
  <h2
    id="title"
  >
    GifExpertApp
  </h2>
  <AddCategory
    categories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
